.github/workflows/docs.yml

```
@@ -0,0 +1,103 @@
+name: Documentation
+
+on:
+  workflow_dispatch:
+  pull_requests:
+  push:
+    branches:
+      - main
+      - develop
+  schedule:
+    # Run every day at 3 am UTC
+    - cron: "0 3 * * *"
+
+env:
+  PYBAMM_DISABLE_TELEMETRY: "true"
+  FORCE_COLOR: 3
+  PYBAMM_IDAKLU_EXPR_CASADI: ON
+  PYBAMM_IDAKLU_EXPR_IREE: ON
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  docs:
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout repository code
+      - name: Check out repository
+        uses: actions/checkout@v4
+
+      # Install Linux system dependencies
+      - name: Install Linux system dependencies
+        uses: awalsh128/cache-apt-pkgs-action@a6c3917cc929dd0345bfb2d3feaf9101823370ad # v1.4.2
+        with:
+          packages: graphviz pandoc
+          execute_install_scripts: true
+
+      # Install TeXLive for Linux
+      - name: Install TeXLive for Linux
+        run: |
+          sudo apt-get update
+          sudo apt-get install texlive-latex-extra dvipng
+
+      # Set up Python
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+          cache: 'pip'
+
+      # Upgrade pip and install dependencies for PyBaMM
+      - name: Install PyBaMM with [docs,all]
+        run: |
+          python -m pip install --upgrade pip
+          pip install -e .[docs,all]
+
+      # Run doctests for Python 3.11
+      - name: Run doctests
+        run: python -m nox -s doctests
+
+      # Build the docs (HTML)
+      - name: Build docs (HTML)
+        run: sphinx-build -b html docs/ docs/_build/html
```

```suggestion
      - name: Build docs (HTML)
        run: sphinx-build -b html -d docs/_build/.doctrees docs docs/_build
```

Building the doctrees with the HTML builder here will mean that they can be reused for the linkcheck builder in the next step.

---

Sorry resolved this by mistake. I re-added the request